src/observers/stores/opentelemetry.py

```diff
@@ -0,0 +1,128 @@
+# stdlib features
+from dataclasses import dataclass
+from typing import Optional
+from importlib.metadata import PackageNotFoundError, version
+
+# Observers internal interfaces
+from observers.observers.base import Record
+from observers.stores.base import Store
+
+# Actual dependencies
+from opentelemetry import trace
+from opentelemetry.sdk.trace import TracerProvider, Tracer, Span
+from opentelemetry.sdk.trace.export import (
+    BatchSpanProcessor,
+)
+from opentelemetry.sdk.resources import Resource
+from opentelemetry.exporter.otlp.proto.grpc.trace_exporter import OTLPSpanExporter
+
+
+OTEL_NAMESPACE = "huggingface.co/observers"
```

Comment:
We also have `observers.dev` that could fit better maybe?  Up to you.

Also would user ever want to change this default?

Reply:
That's a really good question! I considered setting it to `OTEL_SERVICE_NAME` but i think that gets used as a dataset name by otel norms. I could make it one of the things you can set via `connect` tho!

as far as branding, i def wanna go with what's most relevant to the project; do you feel like `observers.dev` is more prominently the brand than `huggingface`?